README.md

```diff
@@ -1 +1,122 @@
+# action-build-and-push-images
 
+This is a composite GitHub action that builds and publishes images to
+
+- GitHub Container Registry (GHCR)
+- Google Artifact Registry (GAR)
+
+## Usage
+
+**Required Permissions**
+Add the following permissions block to your workflow to allow publishing to GHCR and/or GAR:
+```yaml
+permissions:
+  contents: read
+  packages: write      # Required for GHCR
+  id-token: write      # Required for Google Artifact Registry (GAR)
+```
+If you're just building, `packages` should be set to `read` and `id-token` should be omitted.`
+```yaml
+permissions:
+  contents: read
+  packages: read      # Required to read from GHCR if using as cache
+  # id-token is omitted here
+```
+> Adjust permissions as needed for your use case. See [GitHub Actions permissions docs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) for more details.
+
+### Basic Usage
+
+```yaml
+- name: Build and push image
+  uses: getsentry/action-build-and-push-images@sha
+  with:
+    image_name: 'sentry'
+    platform: 'amd64'
+```
+
+### Multi-Architecture with Matrix Strategy
+
+```yaml
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - platform: amd64
+            runner: ubuntu-latest
+          - platform: arm64
+            runner: ubuntu-latest-arm64
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: Build and push image
+        uses: getsentry/action-build-and-push-images@sha
+        with:
+          image_name: 'sentry'
+          platform: ${{ matrix.platform }}
+```
+
+### With Custom Build Configuration
+
+```yaml
+- name: Build and push image
+  uses: getsentry/action-build-and-push-images@sha
+  with:
+    image_name: 'sentry'
+    platform: 'amd64'
+    dockerfile_path: './docker/Dockerfile'
+    build_context: './src'
+    build_target: 'production'
+    build_args: |
+      ENVIRONMENT=production
+      DEBUG=false
+```
+
+### Publishing to Google Artifact Registry
+
+```yaml
+- name: Build and push image
+  uses: getsentry/action-build-and-push-images@sha
+  with:
+    image_name: 'sentry'
+    platform: 'amd64'
+
+    # Enable GAR publishing
+    google_ar: 'true'
+    google_ar_image_name: 'us-central1-docker.pkg.dev/myproject/myrepo/myapp'
```

Comment:
Probably it would be best to link to internal documentation on how to ask for the GAR image repository to be created?

Reply:
I do not think we have documentation around this but there should be. I'll add it when we do have docs on this